src/systems/filecoin_blockchain/struct/block/block.id

```diff
@@ -1,40 +1,39 @@
-import ipld "github.com/filecoin-project/specs/libraries/ipld"
 import filcrypto "github.com/filecoin-project/specs/algorithms/crypto"
 
 import st "github.com/filecoin-project/specs/systems/filecoin_vm/state_tree"
 import clock "github.com/filecoin-project/specs/systems/filecoin_nodes/clock"
 import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
 import msg "github.com/filecoin-project/specs/systems/filecoin_vm/message"
 
-type BlockCID ipld.CID
-type MessageRoot ipld.CID
-type ReceiptRoot ipld.CID
 type ChainWeight UVarint
 type ChainEpoch UVarint
+type MessageReceipt util.Bytes
 
 // On-chain representation of a block header.
 type BlockHeader struct {
     // Chain linking
-    Parents          Tipset
-    Weight           ChainWeight
-    Epoch            ChainEpoch
-
-    // Miner Info
-    MinerAddress     addr.Address
-
+    Parents                [&BlockHeader]
+    ParentWeight           ChainWeight
     // State
-    StateTree        st.StateTree
-    Messages         MessageRoot
-    MessageReceipts  ReceiptRoot
+    ParentStateTree        st.StateTree
+    ParentMessageReceipts  [&MessageReceipt]
```

Comment:
This must be a reference to an array-mapped-trie structure too, not an embedded list.